pkgs/development/python-modules/fastpair/default.nix

```diff
@@ -11,9 +11,9 @@ buildPythonPackage {
     sha256 = "1pv9sxycxdk567s5gs947rhlqngrb9nn9yh4dhdvg1ix1i8dca71";
   };
 
-  buildInputs = [ pytestrunner ];
+  buildInputs = [ (pytestrunner.override { pytest = pytest_3; }) ];
```

Comment:
I think this is in `setup_requires` so this typically would be a `nativeBuildInputs`.

> [Nixpkgs manual Python](https://nixos.org/nixpkgs/manual/#reference)
> nativeBuildInputs ? []: Build-time only dependencies. Typically executables as well as the items listed in setup_requires.

Reply:
I think this is a quirk of `pytestrunner` tending to be included in the `setup_requires` section of `setup.py` when really it's only significant at check-time. The problem with including it in `propagatedBuildInputs` is just that - it means `pytestrunner` and `pytest` will get propagated to all depending packages, which is totally unnecessary and what's more will likely cause collisions between `pytest` and `pytest_3` further down the tree.

Of course, the fun thing is that `pytestrunner` isn't actually used by this derivation - it doesn't seem to have been configured correctly by the upstream author. Hence why we have a manual `checkPhase` below. The only reason it's here in `buildInputs` is to placate `setup.py` which insists it's necessary.

Reply:
>  The problem with including it in `propagatedBuildInputs` is just that - it means `pytestrunner`and `pytest` will get propagated to all depending packages, which is totally unnecessary and what's more will likely cause collisions between `pytest` and `pytest_3` further down the tree (this already happens elsewhere and needs fixing).

I didn't mean to put it in `propagatedBuildInputs` it should just be a `nativeBuildInputs`.
Doing the former would obviously cause collisions, as you stated, and `pytest` isn't a runtime dependency so it wouldn't belong there anyway.

> Of course, the fun thing is that `pytestrunner` isn't actually used by this derivation - it doesn't seem to have been configured correctly by the upstream author. Hence why we have a manual `checkPhase` below. The only reason it's here in `buildInputs` is to placate setup.py which insists it's necessary.

I'm usually partial with just patching it out of `setup_requires` and into `tests_require`.
I think they're just missing aliasing `pytest` to `test` and it would just work without the custom `checkPhase`.

Reply:
Oh I should read peoples comments properly - yeah ok I'll switch it to `nativeBuildInputs` - modernism and all that...

Yeah... I try to avoid patching when I can - it always feels as though it's going to be more fragile than most alternatives.

Reply:
> Yeah... I try to avoid patching when I can - it always feels as though it's going to be more fragile than most alternatives.

Upstream fixes would be best. But we already have, I think, a few patches in use already.